### ISSUE TITLE
[@typescript-eslint/explicit-module-boundary-types] doesn't handle curried functions

### ISSUE THREAD

**Issue by karlismelderis** [typescript-eslint/typescript-eslint#2219]
I might be wrong as I'm still learning TypeScript :)
but if tsc doesn't complain about this bit of code I hope I did type it correctly
```js
type A = (arg: string) => (arg: string) => void

export const a: A = (source) => (destination) => {
  console.error(`${source}:${destination}`)
}
```

**Expected Result**

I hope ESLint rule won't give any error

**Actual Result**

ESLint is giving me two errors:

ESLint: Argument 'destination' should be typed.(@typescript-eslint/explicit-module-boundary-types)
ESLint: Missing return type on function.(@typescript-eslint/explicit-module-boundary-types)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.1` |
| `@typescript-eslint/parser`        | `3.0.1` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.0.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.14.5` |

**Comment by essenmitsosse** [comment]
This bug should be fixed in the latest version. Try updating to 3.2.0

**Comment by bradzacher** [comment]
Please ensure you upgrade to the latest version before raising an issue:
https://github.com/typescript-eslint/typescript-eslint/blob/master/CONTRIBUTING.md#raising-issues

**Comment by karlismelderis** [comment]
Lesson learned :)
